enterprise/server/sociartifactstore/sociartifactstore.go

```diff
@@ -285,112 +327,196 @@ func deserializeDigest(s string) (*repb.Digest, error) {
 
 }
 
-func pullAndIndexImage(ctx context.Context, imageName, imageRef string) (*repb.Digest, []*repb.Digest, error) {
+func (s *SociArtifactStore) pullAndIndexImage(ctx context.Context, imageName string, credentials *rgpb.Credentials) (*repb.Digest, []*repb.Digest, error) {
 	log.Infof("soci artifacts not found, generating them for image %s", imageName)
-	if err := pullImage(ctx, imageName); err != nil {
+	image, err := pullImage(ctx, imageName, credentials)
+	if err != nil {
 		return nil, nil, err
 	}
-	if err := runSoci(ctx, imageName); err != nil {
+	return s.indexImage(ctx, image)
+}
+
+func pullImage(ctx context.Context, image string, credentials *rgpb.Credentials) (v1.Image, error) {
```

Comment:
this function is looking up the image descriptor, it's not pulling the image contents so I'd call it something else

Reply:
Renamed to getTargetImageInfo and changed to return the target image as a ctrname.Digest per your comment below.